Handler/Sites.hs

```diff
@@ -102,11 +102,19 @@ siteForm msite = renderBootstrap3 BootstrapBasicForm $ Site
 
     baseUrlField :: Field Handler Text
     baseUrlField = flip checkM textField $ \url -> do
-        n <- runDB $ count [SiteBaseUrl ==. url]
+        if unchanged url
+            then return $ Right url
+            else ensureUnique url
 
-        return $ case n of
-            0 -> Right url
-            _ -> Left ("This URL is already in use" :: Text)
+      where
+        unchanged url = fmap siteBaseUrl msite == Just url
```

Comment:
I usually avoid this unless since, as you noticed, it can be unclear. I considered it worth it in this case vs naming and passing an explicit argument (e.g. `msite'` or something).